### DISCUSSION TITLE
PRISMA - How to filter the relation nested after m2m relation?

### DISCUSSION THREAD

**Discussion by BGabrielD** [prisma/prisma#16229] [Q&A] [answered]
How to filter a relation nested after m2m relation? 

Schema.graphql:

 ```
  type Entity {
    id: ID!
    name: String
  }

  type Service {
    id: ID!
    service_entity_m2m: [ServiceEntityM2m]
  }

  type ServiceEntityM2m {
    id: ID!
    entity: Entity!
    service: Service!
  }
```

seeder:

// services

    {
      id: 1
    }

// service_entity_m2m:

    {
      id: 1,
      service_id: 1,
      entity_id: 1
    },
    {
      id: 2,
      service_id: 1,
      entity_id: 2
    }

// entities

    {
      id: 1,
      name: "XYZ"
    },
    {
      id: 2,
      name: "Test"
    }

This code:

    context.prisma.service.findMany({
        where: { service_entity_m2m: { some: { entity: { name: { contains: "Test" } } } } },
        include: { service_entity_m2m: { include: { entity: true } } },
    })

returns both entities: 

    [ 
      { 
        id: 1, 
        service_entity_m2m: 
        [
          { 
            id: 1, 
            entity: { 
              id: 1, 
              name: "XYZ" 
            } 
          },
          {
            id: 2, 
            entity: { 
              id: 2, 
              name: "Test" 
            } 
          }
        ]
      }
    ]

How to write a query that is going to return only one entity with the name "Test": ?

    [ 
      { 
        id: 1, 
        service_entity_m2m: 
        [
          {
            id: 2, 
            entity: { 
              id: 2, 
              name: "Test" 
            } 
          }
        ]
      }
    ]

**Comment by pantharshit00** [comment, marked as answer]
Hello @BGabrielD 

You will need to use the following query:
```ts
const data = await prisma.service.findMany({
    where: {
      service_entity_m2m: { some: { entity: { name: { contains: "Test" } } } },
    },
    include: {
      service_entity_m2m: {
        where: {
          entity: {
            name: {
              contains: "Test",
            },
          },
        },
        include: { entity: true },
      },
    },
  });
  ```
  
  Since you also want to filter the nested relationships which you are fetching so you will also need to provide `where` in the `include` field.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
